vision/cloud-client/src/main/java/com/example/vision/Detect.java

```diff
@@ -0,0 +1,405 @@
+  public static void argsHelper(String[] args, PrintStream out) throws IOException{
+    if (args.length < 1) {
+      out.println("Usage:");
+      out.printf(
+          "\tjava %s \"<command>\" \"<path-to-image>\"\n" +
+          "Commands:\n\tall-local | faces | labels | landmarks | logos | text | safe-search | properties\n" +
+          "Path:\n\tA file path (ex: ./resources/wakeupcat.jpg) or a URI for a Cloud Storage resource (gs://...)\n",
+          Detect.class.getCanonicalName());
+      return;
+    }
+    String command = args[0];
+    String path = args.length > 1 ? args[1] : "";
+
+    Detect app = new Detect(ImageAnnotatorClient.create());
+
+    if (command.equals("all-local")) {
+      detectFaces("resources/face_no_surprise.jpg", System.out);
+      detectLabels("resources/wakeupcat.jpg", System.out);
+      detectLandmarks("resources/landmark.jpg", System.out);
+      detectLogos("resources/logos.png", System.out);
+      detectText("resources/text.jpg", System.out);
+      detectProperties("resources/landmark.jpg", System.out);
+      detectSafeSearch("resources/wakeupcat.jpg", System.out);
+    } else if (command.equals("faces")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectFaces(path, System.out);
+      }
+    } else if (command.equals("labels")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectLabels(path, System.out);
+      }
+    } else if (command.equals("landmarks")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectLandmarks(path, System.out);
+      }
+    } else if (command.equals("logos")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectLogos(path, System.out);
+      }
+    } else if (command.equals("text")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectText(path, System.out);
+      }
+    } else if (command.equals("properties")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectProperties(path, System.out);
+      }
+    } else if (command.equals("safe-search")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectSafeSearch(path, System.out);
+      }
+    }
+  }
+
+  private static ImageAnnotatorClient visionApi;
+
+  /**
+   * Constructs a {@link Detect} which connects to the Cloud Vision API.
+   */
+  public Detect(ImageAnnotatorClient client) {
+    visionApi = client;
+  }
+
+  /**
+   * Helper for getting byte array given input stream.
+   * @param is Input stream to get bytes for.
+   * @return Byte array for the input stream contents.
+   * @throws IOException
+   */
+  static public byte[] toByteArray(InputStream is) throws IOException {
```

Comment:
Can you run `mvn checkstyle:checkstyle` in this subproject? I'm pretty sure this should be `public static byte[] ...`